guides/micronaut-intellij-idea-ide-setup/micronaut-intellij-idea-ide-setup.adoc

```diff
@@ -0,0 +1,82 @@
+This guide teaches you how to set up https://www.jetbrains.com/idea/[IntelliJ IDEA] to develop Micronaut applications.
+____
+IntelliJ IDEA – the Leading Java and Kotlin IDE
+The IDE that makes development a more productive and enjoyable experience
+____
+
+== IntelliJ IDEA features for Micronaut Framework
+
+IntelliJ IDEA comes in two flavors https://www.jetbrains.com/idea/download[Ultimate and Community Editions]. You can develop Micronaut applications with the community edition. However, the https://www.jetbrains.com/help/idea/micronaut.html[Ultimate edition has several extra features tailored to the Micronaut Framework].
+
+____
+IntelliJ IDEA provides the following:
+
+* Coding assistance specific to the Micronaut API and the configuration file parameters. For example, when writing query methods, generating HTTP requests for defined endpoints, and so on.
+* Integration with the Bean Validation and Endpoints tool windows.
+* A dedicated project creation wizard based on launch.micronaut.io.
+* A dedicated run configuration for Micronaut applications.
+____
+
+:exclude-for-build:maven
+
+== IntelliJ IDEA and Gradle
+
+You can build Micronaut Framework with https://gradle.org/[Gradle]. The https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/[Micronaut Gradle Plugin] offers a great development experience.
+
+https://www.jetbrains.com/help/idea/gradle.html[IntelliJ IDEA supports Gradle] out of the box both in the community and ultimate editions.
+
+== Build and run actions
+
+https://www.jetbrains.com/help/idea/work-with-gradle-projects.html#delegate_build_gradle[Delegate run and build actions to Gradle]:
+
+image::build-and-run-using-gradle.png[]
+
+____
+If you use the annotationProcessors in your project, we recommend that you delegate run and build actions to Gradle, so the annotation processor will be correctly enabled in your project.
+
+* In the Settings dialog (⌘ ,), go to Build, Execution, Deployment | Gradle.
+* On the Gradle settings page, in the Gradle Projects section, select a Gradle project you need.
+* In the Build and run using list, select the appropriate option and click OK to save the changes.
+____
+
+NOTE: Some features such as https://micronaut-projects.github.io/micronaut-test-resources/latest/guide/[test resources] rely on the Micronaut Gradle Plugin being used to run and test your application.
+
+:exclude-for-languages:groovy
+
+=== Enable annotation Processing
```

Comment:
@alvarosanchez should I show section also in the Maven guide?. I think so right?

Reply:
Correct

Reply:
resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1241/commits/eaefd65081dabd12e3c9a88aafa385b1ce43ca0f